src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java

```diff
@@ -0,0 +1,103 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+import com.es.phoneshop.model.order.DefaultOrderService;
+import com.es.phoneshop.model.order.Order;
+import com.es.phoneshop.model.order.OrderService;
+import com.es.phoneshop.model.order.PaymentMethod;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class CheckoutPageServlet extends HttpServlet {
+
+    protected static final String CHECKOUT_JSP = "/WEB-INF/pages/checkout.jsp";
+
+    private CartService cartService;
+    private OrderService orderService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        cartService = DefaultCartService.getInstance();
+        orderService = DefaultOrderService.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        var cart = cartService.getCart(request);
+        request.setAttribute("order", orderService.getOrder(cart));
+        request.setAttribute("paymentMethods", orderService.getPaymentMethods());
+        request.getRequestDispatcher(CHECKOUT_JSP).forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        var cart = cartService.getCart(request);
+        var order = orderService.getOrder(cart);
+        Map<String, String> errors = new HashMap<>();
+
+        setRequiredParameter(request, "firstName", errors, order::setFirstName);
+        setRequiredParameter(request, "lastName", errors, order::setLastName);
+        setRequiredParameter(request, "phone", errors, order::setPhone);
```

Comment:
Any specific validation for pnone needed? For example, only digits?

Reply:
Added a regular expression check. Now only numbers are accepted